core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -66,6 +69,21 @@ protected ResourcePool(int maxCapacity, ConcurrentLinkedQueue<T> resources) {
    * @return a resource taken from the pool
    */
   public T acquire() {
+    return acquire(null, null);
+  }
+  /**
+   * Acquires an object of type {code T} from the pool.
+   *
+   * This method is like {@link #acquire()}, but it will time out if an object cannot be
+   * acquired before the specified amount of time.
```

Comment:
Is it guaranteed if `mNotEmpty.await` returns, `mResources.poll` will succeed? Otherwise we could wait longer than the specified time right?